feat: remove sismo-connect-solidity library dependencies

yum0e opened the pull request:
- [x] rewrite SismoConnectVerifier tests
- [x] rewrite HydraS3Verifier tests
- [x] remove builder libraries dependencies in tests and deployment scripts (libraries are now deployed in [sismo-core/sismo-connect-solidity](https://github.com/sismo-core/sismo-connect-solidity/pull/3))

yum0e committed:
feat: started removing library dependencies, fix tested for
SismoConnectVerifier

yum0e committed:
fix: tests for HydraS3Verifier + remove libs folder

yum0e committed:
feat: remove hardhat folder

yum0e committed:
feat: remove libraries from the deployment script

yum0e committed:
feat: setup to deploy on base network (#44)

* feat: setup to deploy on base network

* feat: add deployment logs on base-goerli for core contracts

* feat: add deployment logs on base for core contracts

* feat: successfully deployed AddressesProvider contract on Base and
BaseGoerli networks

leosayous21 commented:
In `src/utils` there is some AuthBuilder, ClaimBuilder libraries, etc. that contains a lot of "build" function. I don't think this is necessary as in the code we are always using the same build function. 
I would also put them as an internal library instead of external one

yum0e committed:
feat: remove useless `build` functions for testing purposes libraries
(`AuthBuilder`, `ClaimBuilder` and `ProofBuilder`) + keep remaining as
internals
